src/vs/workbench/contrib/notebook/browser/services/notebookExecutionServiceImpl.ts

```diff
@@ -77,6 +78,8 @@ export class NotebookExecutionService implements INotebookExecutionService, IDis
 
 		// request execution
 		if (validCellExecutions.length > 0) {
+			await this._notebookExecutionStateService.runExecutionParticipants(validCellExecutions);
```

Comment:
Sorry I said to put the participants on the execution _state_ service but didn't realize you are in the `NotebookExecutionService` here, I think the participants should actually be on the `NotebookExecutionService`. Then they are managed by the same thing that calls them and `runExecutionParticipants` won't have to be public.

Reply:
Yep, that definitely makes sense. Looks cleaner overall, and a bit more intuitive.